### ISSUE TITLE
The loss_att is significantly less than other loss

### ISSUE THREAD

**Issue by okkteam** [espnet/espnet#2193]
Hi, recently I noticed that the loss_rnnt is significantly less than other loss when I tried a multi task Transducer mode. For example:  
`iteration` `main/loss_ctc` `main/loss_rnnt` `main/loss_att`   
----100---------637.238---------732.851--------8.422218
----200---------631.205---------615.779--------8.46106                     

So is this a normal phenomenon？

**Comment by sw005320** [comment]
@b-flo, could you answer it?
I'm also confused, but probably `loss_rnnt` is normalized by the length?

**Comment by b-flo** [comment]
Hi,

Hum I'm also confused... Could you give us more information please (espnet version, warp-rnnt commit, dataset used, etc)?
Btw, what is multi-task transducer mode?

**Comment by wxt1997** [comment]
> I'm also confused, but probably `loss_rnnt` is normalized by the length?

Sorry, I don't really understand this normalization and how to located the codes. I only know a normalization by the length during beam search. 

> Hum I'm also confused... Could you give us more information please (espnet version, warp-rnnt commit, dataset used, etc)?
> Btw, what is multi-task transducer mode?

espnet version: master brunch
warp-rnnt: warprnnt-pytorch 0.1 
dataset: aishell
Transducer config: 
```
etype: transformer
dtype: transformer
model_module: "espnet.nets.pytorch_backend.e2e_asr_transudcer:E2E"
rnnt-mode: 'rnnt'
mtlalpha: 0.0  # loss = mtlalpha * loss_ctc + mtlbeta * loss_rnnt + mtlgamma * loss_att 
mtlbeta: 0.5
mtlgamma: 0.5
atype: "add"
opt: noam
criterion: loss
early-stop-criterion: "validation/main/loss"
```
The loss_att is the same as the attention loss in joint-ctc/attention. Although it's an offline operation, it could be used just during training. The additional codes are in `espnet/nets/pytorch_backend/transducer/transformer_decoder.py`. 
And no changes in both loss_rnnt and loss_ctc
```  
loss_rnnt = self.criterion(pred_pad, target, pred_len, target_len)
loss_ctc = self,ctc(hs_pad, pred_len, ys_pad)
```
Both loss_ctc and loss_att are auxiliary tasks during training a transducer, which aims to explore the relationship among them.

**Comment by wxt1997** [comment]
I'm so sorry, I made a mistake in train.log output order. The loss which is significantly small is `loss_att`.

**Comment by b-flo** [comment]
OK, let me understand something first: this is not standard behaviour, what you're trying to do MTL with transducer included. Right?
There are many training and decoding difference between MTL and transducer architecture, it's a bit difficult to pinpoint what would be the reason. Can we access the code somewhere or it's private work?

**Comment by wxt1997** [comment]
> OK, let me understand something first: this is not standard behaviour, what you're trying to do is multi-task learning with transducer included. Right?
> There are many training and decoding difference between default MTL (CTC/Att) and transducer architecture, it's a bit difficult to pinpoint what would be the reason. Can we access the code somewhere or it's private work?
> 
> P.S: I edited your title and swapped rnnt-att values based on your last message.

Yes, I'm trying to do a multi-task learning with transducer included.  
Thanks for your edit! I would like to push the code in my fork repository as soon as possible.

**Comment by b-flo** [comment]
No problem, feel free to ping us when the code is available!
Btw what is the motivation behind MTL with transducer?

**Comment by wxt1997** [comment]
The hint of MTL transducer is from comparing the alignment compution of ctc and rnnt. Both of them are force alignment, so I just want to see what will happen after swapping ctc-rnnt.

**Comment by wxt1997** [comment]
hi, the loss_att is indeed very small, I also found this phenomenon in `e2e_asr` model, the calculation result is similar to the MTL transducer mentioned earlier. But this is how you use it in `e2e_asr` model
```
loss_att *= np.mean([len(x) for x in ys_in]) - 1
```
In other words, multiply loss_att by the mean value of sentence length before using it. May I ask what is the reason for this.
